hashmap/hashmap.mbt

```diff
@@ -73,30 +101,36 @@ pub fn set[K : Hash + Eq, V](self : HashMap[K, V], key : K, value : V) -> Unit {
   if self.capacity == 0 || self.size >= self.growAt {
     self.grow()
   }
-  let mut idx = self.hash(key)
-  let mut entry = (0, key, value) // (distance, key, value)
+  let hash = self.make_hash(key)
+  let mut idx = self.index(hash)
```

Comment:
```suggestion
  loop 0, self.index(hash), 0, hash, key, value {
    i, idx, psl, hash, key, value => {
      if i == self.capacity {
        abort("HashMap is full")
      }
      match self.entries[idx] {
        Empty => {
          self.entries[idx] = Entry::Valid(psl, hash, key, value)
          self.size += 1
          break
        }
        Valid(d, h, k, v) => {
          if h == hash && k == key {
            self.entries[idx] = Valid(d, h, k, value)
            break
          }
          if psl > d {
            self.entries[idx] = Entry::Valid(psl, hash, key, value)
            continue i + 1, self.next_index(idx), d + 1, h, k, v
          }
          continue i + 1, self.next_index(idx), psl + 1, hash, key, value
        }
      }
    }
  }
```

Do you think it is better to write with functional loop?

Reply:
Yes, it is more concise and less error-prone. I will modify other loops.